### DISCUSSION TITLE
RLS Policy for deleting other user's replies that have parent_id associated with answer's id

### DISCUSSION THREAD

**Discussion by Domino-98** [supabase/supabase#6460] [Questions] [answered]
When I delete an answer, I want to be able to delete all associated replies of this answer (same 'answers' table for replies but with 'parent_id' column with id of parent answer). I tried something like `(uid() IN ( SELECT user_id
   FROM answers
  WHERE (answers.parent_id = answers.id)))` but it doesn't work.
Here's example in my table (can't delete 'test's' reply as 'User123')
![1](https://user-images.githubusercontent.com/64014834/163677591-69dd3e33-e905-47cd-9abd-37087792fcdf.JPG)
![2](https://user-images.githubusercontent.com/64014834/163677592-0c7613ee-2710-4b50-b9fc-aa141d9e928f.JPG)

**Comment by GaryAustin1** [comment, marked as answer]
You also need a valid select policy to do the delete.
Something like this worked for me for delete policy where test.number of the current column pointed to another column id with the user_id also matching uid() in that row.
```
(EXISTS ( SELECT 1
   FROM test test2
  WHERE ((test2.id = test.number) AND (test2.user_id = auth.uid()))))
```
You may just need to add an alias to your policy with FROM answers answers2 WHERE (answers.parent_id = answers2.id)
But I've not tested it.

Warning be careful of self-referencing policies especially on select.

**Reply by Domino-98** [reply]
Thanks!
`(EXISTS ( SELECT 1
   FROM answers answers2
  WHERE ((answers.parent_id = answers2.id) AND (answers2.user_id = uid()))))` works for me.
